packages/nitro-protocol/hardhat.config.ts

```diff
@@ -20,11 +20,27 @@ if (!mainnetDeployerPK) {
 
 const config: HardhatUserConfig = {
   solidity: {
-    version: '0.7.4',
-    settings: {
-      optimizer: {
-        enabled: true,
-        runs: 200,
+    compilers: [
+      {
+        version: '0.7.4',
+        settings: {
+          optimizer: {
+            enabled: true,
+            runs: 200,
+          },
+        },
+      },
+    ],
+    overrides: {
+      // This configuration is a workaround for an example contract which doesn't compile with the optimzer on.
+      // The contract is not part of our core protocol.
+      // It is an example of an application a third party dev might write, so it is highly nonideal that it requires this workaround.
+      // See https://github.com/ethereum/solidity/issues/10930
+      'contracts/examples/EmbeddedApplication.sol': {
+        version: '0.7.4',
+        settings: {
+          optimizer: {enabled: false},
+        },
```

Comment:
Nit: This change is actually *not* about monorepo configuration. This is more a problem caused by using the exit format. 

In an ideal world, we would like to revert this commit when we come to tackle #3759; but here we will end up reverting some change that we would likely still need.